### PR TITLE
Deberta v2 code simplification

### PR DESCRIPTION
**Pull request by guillaume-be**
# What does this PR do?

This PR simplifies and fixes the code for the DeBERTa V2 disentangled attention bias calculation:
1. Removes a spurious substraction of the type `x - x` always resulting in 0
2. Fixes condition checking for the attention type and attention score calculation. In the current version, the additional check to see if `p2p` is in the attention type is performed at the wrong position (https://github.com/huggingface/transformers/blob/2c2a31ffbcfe03339b1721348781aac4fc05bc5e/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L781).  The `p2c` attention is not used for `p2p`, but `c2p` is. Currently the execution would fail if the attention types include `p2p` but not `c2p`: the c2p_pos variable used https://github.com/huggingface/transformers/blob/2c2a31ffbcfe03339b1721348781aac4fc05bc5e/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L813 would not be defined at https://github.com/huggingface/transformers/blob/2c2a31ffbcfe03339b1721348781aac4fc05bc5e/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L772. This PR moves the check for the `p2p` attention flag to the right position and simplifies the `p2c` attention calculation.

## Who can review?

@LysandreJik
@BigBird01

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
This looks good to me! Thanks @guillaume-be.

Waiting for a quick review from @BigBird01.

@guillaume-be  please ping me again in ~1 week to merge that PR if it hasn't moved then. Thanks for your contribution!

**Comment by guillaume-be** [comment]
Hello @LysandreJik , @BigBird01 ,
As discussed a couple of weeks back I wanted to reach out to see if any further work would be required on this PR.

Thank you!

File: src/transformers/models/deberta_v2/modeling_deberta_v2.py
Comment (BigBird01) [inline src/transformers/models/deberta_v2/modeling_deberta_v2.py:769]: This line should be keep unchanged. Actually p2p is not used in our current public model.

File: src/transformers/models/sew_d/modeling_sew_d.py
Comment (BigBird01) [inline src/transformers/models/sew_d/modeling_sew_d.py:846]: Same as the above comment.

**Review by BigBird01 — CHANGES_REQUESTED** [review]

File: src/transformers/models/deberta_v2/modeling_deberta_v2.py
Comment (guillaume-be) [reply src/transformers/models/deberta_v2/modeling_deberta_v2.py:769]: If we keep this line unchanged I suggest removing the p2p support entirely (as it is the case for Deberta). As otherwise the variable `c2p_pos` used by the `p2p` attention would not be initialized. I will push an update to this PR make these changes.

**Review by BigBird01 — APPROVED** [review]

**Comment by LysandreJik** [comment]
LGTM, also pinging @anton-l  as it affects SEW/SEW-D

**Comment by guillaume-be** [comment]
Hello @LysandreJik , @anton-l 
I was wondering if there is further work required on this PR?
Thank you!

**Comment by anton-l** [comment]
Hi @guillaume-be, no further changes required from my side (SEW-D models), feel free to merge if everything else is ok 🙂

**Comment by LysandreJik** [comment]
Thanks for your PR, merging!

**Pull request merged**
